### PR TITLE
Working through various issues from OKR / blog use.

### PR DESCRIPTION
**Pull request by jdunkerley**
### Pull Request Description

- Fixed `list_directory` so that it works with defaults and uses the `Folder_Browse`.  
image
- Adjusted types of `order_by` in `Aggregate_Column.First` and `Aggregate_Column.Last` so supports single `Sort_Column`.
- Adjusted types of `order_by` in `Aggregate_Column.First` and `Aggregate_Column.Last` so an empty Vector is valid (treated as Nothing).
- Adjusted `Count_Distinct` so if an empty set of columns then equivalent to `Count`.
- Use the multi-select widget for `columns` in `Count_Distinct`.
image
- Sort the widget for `order_by` in `First` / `Last`.
image
- Bug fix for widgets with both tagValues and dynamic data. (Thanks @farmaazon).
- Use the multi-select for various things.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: distribution/lib/Standard/Database/0.0.0-dev/src/DB_Table.enso [hunk 1]
```diff
@@ -2839,7 +2845,7 @@ type DB_Table
          Replace texts in quotes with parentheses.
 
              table.text_replace ["Input"] '"(.*?)"'.to_regex '($1)'
-    @columns Widget_Helpers.make_column_name_vector_selector
+    @columns (Widget_Helpers.make_column_name_multi_selector use_regex=True)
```
Comment (kazcw) [inline distribution/lib/Standard/Database/0.0.0-dev/src/DB_Table.enso]: `add_regex`

File: distribution/lib/Standard/Database/0.0.0-dev/src/DB_Table.enso [hunk 2]
```diff
@@ -1620,7 +1622,7 @@ type DB_Table
              #    1 | 20 | b | f
              #    2 | 30 | c | g
              #    3 | 40 | d | h
-    @columns Widget_Helpers.make_column_name_vector_selector
+    @columns (Widget_Helpers.make_column_name_multi_selector use_regex=True)
```
Comment (kazcw) [inline distribution/lib/Standard/Database/0.0.0-dev/src/DB_Table.enso]: `add_regex`

**Review by kazcw — APPROVED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso [hunk 3]
```diff
@@ -552,7 +552,7 @@ type Decimal
 
              remainder = Decimal.new -5 . remainder 3
              # => -2
-    remainder : Decimal -> Decimal -> Decimal
```
Comment (AdRiley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso:555]: Were these not the return values?

**Review by AdRiley — APPROVED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso (hunk 3 above)
Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso:555]: Does need the self type in this line.

**Review by GregoryTravis — APPROVED** [review]

**Pull request merged**